src/Classes/Models/EventModel.php

```diff
@@ -78,4 +78,23 @@ public function addEvent(EventEntity $newEvent):bool
         $query->bindParam(':notes', $newEvent->getNotes());
         return $query->execute();
     }
+
+    /**
+     * Search all events from the database 
+     *
+     * @param string of validated search term
+     * @return array An array of Events based on input search criteria
+     */
+    public function searchEvents(string $searchTerm):array
+    {
+        $sql = 'SELECT `events`.`id`, `events`.`name`, `events`.`category`, 
+                `event_categories`.`name` AS `category_name`, `location`, `date`, `start_time`,`end_time`, `notes` 
+                FROM `events` 
+                LEFT JOIN `event_categories` ON `events`.`category` = `event_categories`.`id` 
+                WHERE `events`.`name` LIKE ? ORDER BY `date` DESC;';
+        $query = $this->db->prepare($sql);
+        $searchTerm = '%' . $searchTerm . '%';
+        $query->execute([$searchTerm]);
+        return $query->fetchAll();
```

Comment:
To test this works you may need to do a var_dump as it might not work with the current state of the GetEventsController (which we'll do in c2t4)

Reply:
What happens if the db doesn't exist? Maybe remember this for the next task. Otherwise this is all good.

Reply:
This is a problem for task 5.

Reply:
ok :)

Reply:
fine

Reply:
Okay, that's fine.